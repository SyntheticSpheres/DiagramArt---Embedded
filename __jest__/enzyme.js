// https://github.com/airbnb/enzyme/issues/1265

const Enzyme = require('enzyme');
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
