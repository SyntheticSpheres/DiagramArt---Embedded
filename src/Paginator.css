.da-pagination ul {
    list-style: none;
    padding: 0;
    margin: 0;
}

.da-pagination ul li {
    display: inline-block;
    margin-right: 5px;
}

.da-pagination ul li.active a{
    background-color: #b8cb3d;
}

.da-pagination ul li a {
    display: block;
    background-color: rgb(70, 78, 84);
    color: white;
    padding: 6px 15px;
    cursor: pointer;
}

.da-pagination ul li a:hover {
    background-color: #b8cb3d;
}

.da-pagination ul li a.active {
    background-color: #b8cb3d;
    cursor: default;
}
