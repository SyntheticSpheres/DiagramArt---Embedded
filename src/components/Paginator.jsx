import React from 'react';

/**
 *
 * @author Matej 'juffalow' Jellus <juffalow@juffalow.com>
 */
class Paginator extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        var pages = [];

        if( this.props.page > 1 ) {
            pages.push(<li key="first">First</li>);
        }

        pages.push(<li key={this.props.page}><a role="button">{this.props.page}</a></li>);

        if( this.props.page < this.props.total ) {
            pages.push(<li key="last"><a role="button">Last</a></li>);
        }

        return (
            <ul>
                {pages}
            </ul>
        );
    }
}

export default Paginator;
