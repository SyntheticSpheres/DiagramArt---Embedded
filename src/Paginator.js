import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Paginator.css';

const Page = ({page, onClick, active}) => ( <li className={active && "active"}><a role="button" onClick={onClick}>{page}</a></li> );

class Paginator extends Component {
  constructor(props) {
    super(props);

    this.onClick = this.onClick.bind(this);
  }

  onClick(page) {
    this.props.onPageChange(page);
  }

  render() {
    const pages = [];

    // first page
    this.props.page > 1 && pages.push(<Page key={1} page={1} onClick={this.onClick.bind(this, 1)} />);
    // page previous page
    this.props.page - 1 > 1 && pages.push(<Page key={this.props.page - 1} page={this.props.page - 1} onClick={this.onClick.bind(this, this.props.page - 1)} />);
    // actual page
    pages.push(<Page key={this.props.page} page={this.props.page} onClick={this.onClick.bind(this, this.props.page)} active={true} />);
    // next page
    this.props.page + 1 < this.props.pages && pages.push(<Page key={this.props.page + 1} page={this.props.page + 1} onClick={this.onClick.bind(this, this.props.page + 1)} />);
    // last page
    this.props.page < this.props.pages && pages.push(<Page key={this.props.pages} page={this.props.pages} onClick={this.onClick.bind(this, this.props.pages)} />);

    return (
      <ul>
        {pages}
      </ul>
    );
  }
}

Paginator.propTypes = {
  page: PropTypes.number.isRequired,
  pages: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
}

export default Paginator;
