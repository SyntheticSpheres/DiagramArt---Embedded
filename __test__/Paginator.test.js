import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import Paginator from '../src/Paginator';

describe('Paginator component', () => {
  let sandbox;
  let callback;

  beforeEach(function() {
    sandbox = sinon.sandbox.create();
    callback = sandbox.stub();
  });

  afterEach(function() {
    sandbox.restore();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Paginator page={1} pages={3} onPageChange={callback} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Paginator page={3} pages={5} onPageChange={callback} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('simlate on page click', () => {
    const tree = shallow(
      <Paginator page={3} pages={5} onPageChange={callback} />
    );
    tree.find({ page: 1 }).simulate('click');
    sinon.assert.calledOnce(callback);
  });
});
