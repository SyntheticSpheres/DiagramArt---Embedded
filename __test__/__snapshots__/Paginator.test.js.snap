// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component renders correctly 1`] = `
<ul>
  <li
    className={undefined}
  >
    <a
      onClick={[Function]}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className={undefined}
  >
    <a
      onClick={[Function]}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="active"
  >
    <a
      onClick={[Function]}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className={undefined}
  >
    <a
      onClick={[Function]}
      role="button"
    >
      4
    </a>
  </li>
  <li
    className={undefined}
  >
    <a
      onClick={[Function]}
      role="button"
    >
      5
    </a>
  </li>
</ul>
`;
